### ISSUE TITLE
AbstractControl enable/disable does not pass emitEvent to this._updateAncestors()

### ISSUE THREAD

**Issue by pkoral** [angular/angular#12366]
**I'm submitting a ...**  (check one with "x")

```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request
```

**Current behavior**

Enabling/disabling `AbstractControl` descendant emits new value in `valueChanges` stream regardless value of `emitEvent` passed in object as parameter of `enable()` / `disable()`.

**Expected behavior**

No new object emitted by valueChanges.

**Minimal reproduction of the problem with instructions**

Create form with 2 controls. Subscribe to form valueChanges stream. In this function disable second control based on value of first control.
`this.control2.disable({ emitEvent: false })`

This causes valueChanges to emit next object.

**What is the motivation / use case for changing the behavior?**

**Please tell us about your environment:**

- **Angular version:** 2.0.X, 2.1.0
  
  
- **Browser:** Chrome XX | Firefox XX
  
  
- **Language:** TypeScript 2.0.3
- **Node (for AoT issues):** `node --version` =   

**Similar issues**

https://github.com/angular/angular/pull/11949

**Comment by DzmitryShylovich** [comment]
Can u provide reproduction in plunkr?

**Comment by koral00** [comment]
Here is the plunkr
https://plnkr.co/edit/YTdbKDYP3UbQmQ9wODpM?p=preview
show browser's debug console, type `1` into first input field.

**Comment by zakton5** [comment]
Any updates on this? As said above, this causes all valueChanges subscriptions within the disabled form group to emit an event.

**Comment by jculverwell** [comment]
I can confirm I am also setting this issue.

**Comment by megan-starr9** [comment]
confirm also, this is currently causing a breaking change on our application due to a valueChanges feedback loop
(commenting to follow this)

**Comment by Bulan** [comment]
Just spent a lot of hours figuring out why valueChanges was triggered on my FormControl, when I changed the disable/enable status of my Form.

Here's another Plunker using latest Angular version
https://plnkr.co/edit/RgyDItYtEfzlLVB6P5f3?p=preview

Created this q on SO
http://stackoverflow.com/questions/43541897/valuechanges-on-formcontrol-triggers-when-form-enable-even-with-emitevent-fals

**Comment by zakton5** [comment]
+1. Any word on this?

**Comment by guyjstitt** [comment]
Also seeing this issue. We are wanting to use Reactive forms in our large application that has complex and nested forms and I consider this a blocking bug.

**Comment by Sleeper9** [comment]
I also encountered this bug when I tried to disable field during async validation process, it triggered an endless flow of valueChanges / disable-enable events.

**Comment by volpejoaquin** [comment]
I also have the same error, please we need a response.

**Comment by guyjstitt** [comment]
I'm also seeing an issue where enabling/disabling a nested form group with emitEvent = false, enables/disables all of the children without passing emitEvent to the children. 

In my application, the child controls of that nested form group are subscribing to valueChanges of their sibling controls. So when the parent form group is enabled, there is an event emitted for each child control. This is breaking the enable/disable logic I have, because we don't want the form group's child controls to update/enable/disable, just because I enabled their form group.

I will create a separate issue for this with details and plunker.

**Comment by elishowk** [comment]
I wonder why emitEvent is not passed to the children here [in disable()](https://github.com/angular/angular/blob/c59c390cdcd825cca67a422bc8738f7cd9ad42c5/packages/forms/src/model.ts#L318) and here [in enable()](https://github.com/angular/angular/blob/c59c390cdcd825cca67a422bc8738f7cd9ad42c5/packages/forms/src/model.ts#L339)

Please anyone capable to explain to us if this is normal or this is one alledged reason of this issue ? Thanks

**Comment by guyjstitt** [comment]
^^ exactly what I was trying to describe. I don't know if it's intended behavior or not. I was in the midst of creating the plunkr last night, but got caught up with other work. It's easy to replicate though. Just enable a form group and pass emitEvent: false, then log the valueChanges and statusChanges of the controls inside the group.

**Comment by pantonis** [comment]
Please provide a solution. This bug along with the disabled attribute not working for reactive forms gives me a deadlock

**Comment by Mourdraug** [comment]
Only workaround I found was to set flag before enabling or disabling any field and checking for it on valueChanges. Quite annoying.

**Comment by xavs** [comment]
This is a quite annoying bug... Will it be fixed anytime soon?

**Comment by xavs** [comment]
I actually got it to work by using 
`.enable({ onlySelf: true, emitEvent: false })`
instead of 
`.enable({ emitEvent: false })`

**Comment by alobban** [comment]
@xavs I don't understand why passing this object signature works , I used your **.disable({ onlySelf: true, emitEvent: false })**

**Comment by jblankenhorn** [comment]
Please fix!!

**Comment by plvice** [comment]
Up.
Fix will be appreciated.

**Comment by Bulan** [comment]
Got an answer/workaround on SO:
http://stackoverflow.com/questions/43541897/valuechanges-on-formcontrol-triggers-when-form-enable-even-with-emitevent-fals

**Comment by TylerHelmuth** [comment]
Just ran in to this problem as well.  The workaround of adding `onlySelf: true` to `.enable` and `.disable` works, but it feels like I shouldn't have to do that.

**Comment by jondmcelroy** [comment]
Just ran into the issue. Unfortunately the disable {onlySelf:true, emitEvent:false} workaround does not seem to work in the case in which we are calling disable the entire FormGroup and we are watching valueChanges on the FormGroups controls. It will emit the event to the controls.

**Comment by dandouglas** [comment]
I am still getting this issue on Angular 5.2.0 when trying to set the value of a form control dynamically.
I have tried the following
`this.myFormControl.setValue(newValue, {onlySelf: true, emitEvent: false});`
but it still fires the valueChanges event.

**Comment by benelliott** [comment]
@dandouglas It was fixed in 6.0.0.

**Comment by sut123** [comment]
I found a related issue to this in the latest version of Angular. If you watch value changes using this.form.get('controlName').valueChanges instead of this.form.controls['controlName'].valueChanges, it is still emitting the event. I would expect the behavior for those to be the same.
(This was also regardless of whether or not I was using onlySelf: true.)
